data/campaigns/The_Hammer_of_Thursagan/utils/macros.cfg

```diff
@@ -4,30 +4,55 @@
 Mage,Red Mage,White Mage,Arch Mage,Great Mage,Mage of Light,Silver Mage #enddef
 
 #define CONDITIONAL_MAGE_RECRUITING
-    [if]
-        [variable]
-            name=mage_disallow
-            boolean_equals=false
-        [/variable]
-        [not]
-            [have_unit]
-                side=1
-                type={MAGES}
-            [/have_unit]
-        [/not]
-        [then]
-            [message]
-                speaker=narrator
-                message= _ "Without magic users from Master Perrin’s academy in your party, you can no longer recruit new ones."
-                image=wesnoth-icon.png
-            [/message]
-            [disallow_recruit]
-                side=1
-                type=Mage
-            [/disallow_recruit]
-            {VARIABLE mage_disallow true}
-        [/then]
-    [/if]
+    [event]
+        name=die
+        first_time_only=no
+        [filter]
+            side=1
+            type_tree=Mage
```

Comment:
I'll be using it more. But only where I bump into it. Not going back to work already done.